eclipse-scout-core/src/desktop/outline/pages/js/JsPageHelper.ts

```diff
@@ -0,0 +1,470 @@
+    this._uninstallPageWithTableTableListeners(this.page?.detailTable);
+  }
+
+  protected _onNodesInserted(event: TreeNodesInsertedEvent) {
+    if (event.parentNode !== this.page) {
+      return;
+    }
+
+    // The event is triggered twice during the load/reload of a PageWithTable that creates its child pages using the JsPageHelper.
+    // 1. When the child pages are inserted on the UI server it is called for the first time.
+    //    In this case e.g. the text may not be correct as the page was not updated from the table row yet.
+    //    If this happens The PageWithTable is marked with childrenLoaded=false.
+    // 2. The second time this event is triggered is when the PageWithTable replaces its rows.
+    //    Now the child page is ready to be inserted as it was linked to the row and updated from it.
+    //    As the table load/reload is now completed the PageWithTable is marked with childrenLoaded=true.
+
+    // Mark all child pages loading iff the PageWithTable has not loaded its children.
+    for (const childPage of event.nodes) {
+      childPage.setCssClass(styles.toggleCssClass(childPage.cssClass, 'js-page-child-page-loading', !this.page.childrenLoaded));
+      childPage._decorate();
+    }
+  }
+
+  protected _installPageWithTableTableListeners(table: Table) {
+    if (!table) {
+      return;
+    }
+    table.on('rowsInserted', this._tableRowsInsertHandler);
+  }
+
+  protected _uninstallPageWithTableTableListeners(table: Table) {
+    if (!table) {
+      return;
+    }
+    table.off('rowsInserted', this._tableRowsInsertHandler);
+  }
+
+  protected _onTableRowsInserted(event: TableRowsInsertedEvent) {
+    // When rows are inserted in a PageWithTable, all child pages are ready, i.e. they are linked to their row and updated from it.
+    // Send a nodesChanged event in order to send summary information from parent PageWithTable to the UI server.
+    // This information is needed when the browser is reloaded as the child pages will still be available but the table data won't.
+    const pages = event.rows.map(row => row.page).filter(Boolean);
+    (this.page.outline.modelAdapter as OutlineAdapter).sendNodesChanged(pages);
+  }
+
+  /**
+   * Calls loadChildPages on the UI server for the given ids or {@link PageParamDo}s. The pages are created by the Java pendant of the JsPage (see org.eclipse.scout.rt.client.ui.desktop.outline.pages.js.AbstractJsPage.createChildPage).
+   * The returned {@link Promise} is resolved when all child pages are created on the UI server.
+   *
+   * @param idsOrPageParams ids or {@link PageParamDo}s that are used to create child pages on the UI server.
+   * @param replace Whether all child pages shall be replaced. If not given the child pages will always be replaced except for the first call in order to keep existing child pages after a browser reload.
+   */
+  async callLoadChildPages(idsOrPageParams: string | PageParamDo | (string | PageParamDo)[], replace?: boolean): Promise<void> {
```

Comment:
Since the method name is already in plural, we can remove the singular argument types (the caller can easily create an array themselves).


```suggestion
  async callLoadChildPages(idsOrPageParams: (string | PageParamDo)[], replace?: boolean): Promise<void> {
```